packages/smooth_app/lib/database/dao_product.dart

```diff
@@ -130,4 +153,29 @@ class DaoProduct extends AbstractSqlDao implements BulkDeletable {
         json.decode(encodedJson) as Map<String, dynamic>;
     return Product.fromJson(decodedJson);
   }
+
+  Future<String> getFreshLocalDataBase() async {
+    try {
+      final List<String> barcodes = await getAllKeys();
+      if (barcodes.isEmpty) {
+        return 'List is Empty\nNothing to Refresh';
+      }
+      final ProductListQueryConfiguration configuration =
```

Comment:
No way: this is a stupid dao, that just reads and write. If you need to do something more refined, ok but just don't do it there.